src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -45,16 +66,26 @@ public void process(String blobName, String containerName) {
             byte[] rawBlob = tryToDownloadBlob(blobClient);
 
             dispatcher.dispatch(blobName, rawBlob, containerName);
+            UUID envelopeId = envelopeRepository.insert(createNewEnvelope(
```

Comment:
We should first check if the envelope already exists. Only if it doesn't, we should acquire lease on the file and carry on. Otherwise, we'd be flooded by "lease already acquired" cases, which we can avoid.

Reply:
it's done above

Reply:
aaa, ok, my bad